pkg/builder/controller.go

```diff
@@ -157,28 +173,31 @@ func (blder *Builder) Build(r reconcile.Reconciler) (controller.Controller, erro
 
 func (blder *Builder) doWatch() error {
 	// Reconcile type
-	src := &source.Kind{Type: blder.apiType}
+	src := &source.Kind{Type: blder.apiType.item}
 	hdler := &handler.EnqueueRequestForObject{}
-	err := blder.ctrl.Watch(src, hdler, blder.predicates...)
+	allPredicates := append(blder.globalPredicates, blder.apiType.predicates...)
+	err := blder.ctrl.Watch(src, hdler, allPredicates...)
 	if err != nil {
 		return err
 	}
 
 	// Watches the managed types
 	for _, obj := range blder.managedObjects {
-		src := &source.Kind{Type: obj}
+		src := &source.Kind{Type: obj.item}
 		hdler := &handler.EnqueueRequestForOwner{
-			OwnerType:    blder.apiType,
+			OwnerType:    blder.apiType.item,
 			IsController: true,
 		}
-		if err := blder.ctrl.Watch(src, hdler, blder.predicates...); err != nil {
+		allPredicates := append(blder.globalPredicates, obj.predicates...)
```

Comment:
this needs to be a copy first, otherwise you'll end up growing the list with other objects' predicates.

Reply:
(or, rather, you could, depending on what Go decides to do with the slice)

Reply:
you are right I missed it